traefik.go

```diff
@@ -111,6 +117,75 @@ Complete documentation is available at https://traefik.io`,
 
 	traefikConfiguration.ConfigFile = toml.ConfigFileUsed()
 
+	var kv *staert.KvSource
+	var err error
+	storeConfig := &store.Config{
+		ConnectionTimeout: 30 * time.Second,
+		Bucket:            "traefik",
+	}
+	if traefikConfiguration.Consul != nil {
```

Comment:
Maybe the following part should be delegated in each providers in a method `CreateKvSource()`. It would avoid the if/else. WDYT?